code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -42,10 +42,16 @@
 	var/progress_amount = 1
 	if(SSxevolution)
 		progress_amount = SSxevolution.get_evolution_boost_power(hive.hivenumber)
-	var/ovipositor_check = (hive.allow_no_queen_actions || hive.evolution_without_ovipositor || (hive.living_xeno_queen && hive.living_xeno_queen.ovipositor) || caste?.evolve_without_queen)
-	if(caste && caste.evolution_allowed && evolution_stored < evolution_threshold && ovipositor_check)
-		evolution_stored = min(evolution_stored + progress_amount, evolution_threshold)
+	var/ovipositor_check = (hive.allow_no_queen_actions || hive.evolution_without_ovipositor || (hive.living_xeno_queen && hive.living_xeno_queen.ovipositor))
+	if(caste && caste.evolution_allowed && (ovipositor_check || caste?.evolve_without_queen))
 		if(evolution_stored >= evolution_threshold)
+			if(!got_evolution_message)
+				evolve_message()
+				got_evolution_message = TRUE
+			if(ovipositor_check)
+				evolution_stored += progress_amount
+		else
+			evolution_stored += progress_amount
 			evolve_message()
```

Comment:
```suggestion
```
So. Much. Spam. Poor ears of every larva. (Yes, larvas have ears for the purposes of this comment.)

Reply:
suggestion applied